### PR TITLE
[hunspell] upgrade hunspell to 1.7.2

### PR DESCRIPTION
**Pull request by lenaschoenburg**
Upgrades hunspell to [version 1.7.2](https://github.com/hunspell/hunspell/releases/tag/v1.7.2).

The unistd patch does not seem to be necessary anymore: https://github.com/hunspell/hunspell/pull/811

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by oleschoenburg** [comment]
@microsoft-github-policy-service agree

**Comment by JonLiu1993** [comment]
Please consider migrating it to the new functions:
````
vcpkg_copy_tool_dependencies -> [vcpkg_copy_tools](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_copy_tools.md)
````

**Comment by oleschoenburg** [comment]
@JonLiu1993 Okay, I can look into it later today :+1:

**Comment by oleschoenburg** [comment]
@JonLiu1993 I tried to replace `vcpkg_copy_tool_dependencies` with `vcpkg_copy_tools`, could you take another look?

**Comment by dg0yt** [comment]
> Please consider migrating it to the new functions:
> 
> ```
> vcpkg_copy_tool_dependencies -> [vcpkg_copy_tools](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_copy_tools.md)
> ```

I already denied a similar request multiple times. It is always related to autotools-based ports.
`vcpkg_copy_tool_dependencies` is deprecated with regard to tools installed in `bin`. I don't think this also apply for tools which are already in `tools`. Or shall we start to copy/move tools from `tools/${PORT}/bin`  to  `tools/${PORT}`?

**Comment by JonLiu1993** [comment]
This is error log, please take a look:
````
Call Stack (most recent call first):
  ports/hunspell/portfile.cmake:62 (vcpkg_copy_tools)
  scripts/ports.cmake:147 (include)


CMake Error at scripts/cmake/vcpkg_copy_tools.cmake:9 (message):
  TOOL_NAMES must be specified.
Call Stack (most recent call first):
  ports/hunspell/portfile.cmake:62 (vcpkg_copy_tools)
  scripts/ports.cmake:147 (include)
````

**Comment by JonLiu1993** [comment]
> @JonLiu1993 I tried to replace `vcpkg_copy_tool_dependencies` with `vcpkg_copy_tools`, could you take another look?
> 
> I'm not sure that the usage of `vcpkg_copy_tools` is correct. The hunspell port has a "tools" feature - should `vcpkg_copy_tools` only run if that feature is enabled?

You can add an if("tools" IN_LIST FEATURES) function as a judgment in front.

**Comment by JonLiu1993** [comment]
> > Please consider migrating it to the new functions:
> > ```
> > vcpkg_copy_tool_dependencies -> [vcpkg_copy_tools](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_copy_tools.md)
> > ```
> 
> I already denied a similar request multiple times. It is always related to autotools-based ports. `vcpkg_copy_tool_dependencies` is deprecated with regard to tools installed in `bin`. I don't think this also apply for tools which are already in `tools`. Or shall we start to copy/move tools from `tools/${PORT}/bin` to `tools/${PORT}`?

If we only need to move or copy some tools-related dependencies, we only need to use `vcpkg_copy_tool_dependencies`, if we need to move tools from `tools/${PORT}/bin` to `tools/${PORT}` we need to use `vcpkg_copy_tools`, if these tools themselves have Under the `tools/${PORT}` path, we no longer need to use `vcpkg_copy_tools`.

**Comment by JonLiu1993** [comment]
@oleschoenburg, I tried to test the feature twice locally and failed:
````
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: F:/Feature-test/vcpkg/downloads/tools/msys2/ca907ea16f6f2ba5/usr/bin/make.exe -j 9 --trace -f Makefile all
    Working Directory: F:/Feature-test/vcpkg/buildtrees/hunspell/x86-windows-rel/
    See logs for more information:
      F:\Feature-test\vcpkg\buildtrees\hunspell\build-x86-windows-rel-out.log
      F:\Feature-test\vcpkg\buildtrees\hunspell\build-x86-windows-rel-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_make.cmake:149 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_make.cmake:2 (vcpkg_build_make)
  ports/hunspell/portfile.cmake:46 (vcpkg_install_make)
  scripts/ports.cmake:147 (include)
````
[build-x86-windows-rel-out.log](https://github.com/microsoft/vcpkg/files/10566341/build-x86-windows-rel-out.log)
[build-x86-windows-rel-err.log](https://github.com/microsoft/vcpkg/files/10566342/build-x86-windows-rel-err.log)

File: ports/hunspell/vcpkg.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "hunspell",
-  "version": "1.7.1",
+  "version": "1.7.2",
   "port-version": 2,
```
Comment (JonLiu1993) [inline ports/hunspell/vcpkg.json:4]:
```suggestion
```

Comment (JonLiu1993) [reply ports/hunspell/vcpkg.json:4]: We should remove "port-version" when we update the port version.

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by JonLiu1993** [comment]
I tested the feature nls successfully in the x64-osx triplet, but I tested feature tools failed again:
````
-- Building x86-windows-rel
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: F:/Feature-test/hunspell/vcpkg/downloads/tools/msys2/ca907ea16f6f2ba5/usr/bin/make.exe -j 9 --trace -f Makefile all
    Working Directory: F:/Feature-test/hunspell/vcpkg/buildtrees/hunspell/x86-windows-rel/
    See logs for more information:
      F:\Feature-test\hunspell\vcpkg\buildtrees\hunspell\build-x86-windows-rel-out.log
      F:\Feature-test\hunspell\vcpkg\buildtrees\hunspell\build-x86-windows-rel-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_make.cmake:149 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_make.cmake:2 (vcpkg_build_make)
  ports/hunspell/portfile.cmake:46 (vcpkg_install_make)
  scripts/ports.cmake:147 (include)


error: building hunspell:x86-windows failed with: BUILD_FAILED
error: Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
    https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+hunspell
You can submit a new issue at:
    https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[hunspell]+Build+error
Include '[hunspell] Build error' in your bug report title, the following version information in your bug description, and attach any relevant failure logs from above.
    vcpkg-tool version: 2023-01-24-8a88d63f241d391772fbde69af9cab96c3c64c75
    vcpkg-scripts version: 91f5cd371 2023-01-31 (6 days ago)

````
[build-x86-windows-rel-out.log](https://github.com/microsoft/vcpkg/files/10671103/build-x86-windows-rel-out.log)
[build-x86-windows-rel-err.log](https://github.com/microsoft/vcpkg/files/10671104/build-x86-windows-rel-err.log)

**Comment by oleschoenburg** [comment]
@JonLiu1993 Thanks for testing! Unfortunately I don't know how to solve this and I don't even have a windows machine to test this on so I'll close this PR and hope that someone else will upgrade to 1.7.2 in the future :pray:

**Comment by JonLiu1993** [comment]
@oleschoenburg, Thanks for your PR, you can commit a issue in vcpkg for update hunspell to latest version, then we can update it.

**Pull request closed without merging**